<?xml version="1.0"?>
<launch>
    <arg name="image"           default="/image_rect"/>
    <arg name="imu_data"        default="/imu_data"/>
    <arg name="zoom_factor"     default="1.3"/>
    <arg name="enable_trimming" default="true"/>
    <arg name="line_delay"      default="0.0"/>
    <arg name="offset_time"     default="0.0"/>
    <arg name="lsm_order"       default="1"/>
    <arg name="lsm_period"      default="1.5"/>
    <arg name="verbose"         default="false"/> <!-- For Debug -->
    <arg name="allow_blue_space" default="false"/>

    <node name="virtualgimbal_ros" pkg="virtualgimbal_ros" type="line_delay_estimator_node" output="screen">
        <rosparam command="load"    file="$(find virtualgimbal_ros)/params/detector_params.yaml"/>
        <rosparam command="load"    file="$(find virtualgimbal_ros)/params/marker_params.yaml"/>
        <param name="image"             value="$(arg image)"/>
        <param name="imu_data"          value="$(arg imu_data)"/>
        <param name="zoom_factor"       value="$(arg zoom_factor)"/>
        <param name="enable_trimming"   value="$(arg enable_trimming)"/>
        <param name="line_delay"        value="$(arg line_delay)"/>
        <param name="offset_time"       value="$(arg offset_time)"/>
        <param name="lsm_order"         value="$(arg lsm_order)"/>
        <param name="lsm_period"        value="$(arg lsm_period)"/>
        <param name="verbose"           value="$(arg verbose)"/>
        <param name="allow_blue_space"  value="$(arg allow_blue_space)"/>
    </node>
</launch>